lib/archethic/reward.ex

```diff
@@ -47,9 +48,13 @@ defmodule Archethic.Reward do
       |> OracleChain.get_uco_price()
       |> Keyword.get(:usd)
 
-    trunc(50 / uco_usd_price / Calendar.ISO.days_in_month(date.year, date.month) * @unit_uco)
+    number_of_reward_occurences_per_month = number_of_reward_occurences_per_month()
+
+    trunc(50 / uco_usd_price / number_of_reward_occurences_per_month * @unit_uco)
   end
 
+  defp number_of_reward_occurences_per_month(), do: @number_of_reward_occurences_per_month
```

Comment:
What is the purpose to use a private function while you could use directly the module attribute?

Reply:
because reading module attribute directly will run the function at runtime, however if we wrap in a function it will be interpreted at compile time and only the value of the call will be written in the corresponding Beam file.
Since this function may take some time (more than 1s) and the cron job will not change in runtime I thought it's better this way.

the first paragraph of this article explains this a little better. https://medium.com/multiverse-tech/how-to-speed-up-your-elixir-compile-times-part-1-understanding-elixir-compilation-64d44a32ec6e